os/net/routing/rpl-classic/rpl-icmp6.c

```diff
@@ -674,12 +675,21 @@ dao_input_storing(void)
 
   buffer = UIP_ICMP_PAYLOAD;
   buffer_length = uip_len - uip_l3_icmp_hdr_len;
+  if(buffer_length < 1) {
+    LOG_WARN("dropping incomplete DAO\n");
+    return;
+  }
+  last_valid_pos = buffer_length - 1;
 
   pos = 0;
-  instance_id = buffer[pos++];
 
-  instance = rpl_get_instance(instance_id);
+  if(last_valid_pos < 4) {
```

Comment:
Move this before `pos = 0` so input checks are done before the "useful" code starts.